components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -812,7 +821,7 @@ private String resolveRegistryURI(String key) {
      * @throws IOException
      */
     private OMNode readNonXML (URL url) throws IOException {
-        String mediaType = lookUpFileMediaType(url.getPath());
+        Properties metadata = getMetadata(url.getPath());
```

Comment:
Can InputStream **inputStream** = urlConnection.getInputStream() be null ?
If so, shall we move this to line 833. 
If it is null, we don't need to do this, since we are returning null. 
If it is not null , we can remove the null check there.
Correct me if I'm wrong.

Reply:
Will move it below just before the metadata variable is used.